src/lazy/binary/encoded_value.rs

```diff
@@ -88,6 +90,8 @@ pub(crate) struct EncodedValue<HeaderType: EncodedHeader> {
     pub annotations_encoding: AnnotationsEncoding,
     // The offset of the type descriptor byte within the overall input stream.
     pub header_offset: usize,
+    // If this value was written with a tagless encoding, this will be 0. Otherwise, it's 1.
+    pub opcode_length: u8,
```

Comment:
Can this be derived (i.e. `fn is_tagless() -> bool`) rather than storing the opcode length in a field? Tagless encodings _must_ have `value_body_length + length_length == total_length`, and anything else _must_ have an opcode, so `value_body_length + length_length < total_length` is always true for anything that is not a tagless encoded argument. Right?